src/Symfony/Component/Workflow/Dumper/GraphvizDumper.php

```diff
@@ -26,12 +29,17 @@
  */
 class GraphvizDumper implements DumperInterface
 {
+    use GetMetadataTrait;
+
     protected static $defaultOptions = array(
         'graph' => array('ratio' => 'compress', 'rankdir' => 'LR'),
         'node' => array('fontsize' => 9, 'fontname' => 'Arial', 'color' => '#333333', 'fillcolor' => 'lightblue', 'fixedsize' => 'false', 'width' => 1),
         'edge' => array('fontsize' => 9, 'fontname' => 'Arial', 'color' => '#333333', 'arrowhead' => 'normal', 'arrowsize' => 0.5),
     );
 
+    /** @var MetadataStoreInterface */
+    protected $workflowMetadata;
```

Comment:
This class is a service. You should not store specific data in it.

Reply:
Could you please suggest a better solution? I don't know how to fix this, using an interface maybe?

Reply:
you just need to pass the workflow metadata to all method that need to use it

Reply:
Thanks, the code was simpler after removing this. :+1: